crates/orderbook/src/orderbook.rs

```diff
@@ -243,13 +276,58 @@ fn set_available_balances(orders: &mut [Order], cache: &SolvableOrdersCache) {
     }
 }
 
+async fn filter_tokens_without_native_prices(
```

Comment:
Maybe `retain_orders_with_native_prices()` would be better?

Reply:
Here I prefer `filter_` just because the signature is `(Vec<Order>, NativePriceEstimating) -> (Vec<Order>, BTreeSet<H160, U256>)`. A name of `retain_` would suggest to me a signature of `(&mut Vec<Order>, NativePriceEstimating) -> BTreeSet<H160, U256>`.

I'm fine with renaming it, but then I would also suggest changing the signature as well to better match the name.

Reply:
I see your point.
Naming this function is a little hard because it does multiple things:
It filters orders (1) based on their tokens (2) and also returns prices (3).

Maybe `get_orders_with_native_prices` could also be an option?
`get` => to avoid the signature argument (because the current signature is fine IMO)
`orders` => because it operates on orders instead of `tokens`
`with_native_prices` => can be interpreted as "only orders which have native prices" or "return orders but also prices" (both interpretations seem fine to me)

Reply:
I like it! Man, naming things really is the hardest part of software development...

Reply:
Applied your suggestion.